sale_product_set/models/res_partner.py

```diff
@@ -0,0 +1,16 @@
+from odoo import models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    def write(self, vals):
+        res = super().write(vals)
+        if "active" in vals:
+            partner_product_sets = (
+                self.env["product.set"]
+                .with_context(active_test=False)
+                .search([("partner_id", "=", self.id)])
```

Comment:
```suggestion
                .search([("partner_id", "in", self.ids)])
```
You should update your test because ATM this is broken :wink: